src/Kros.AspNetCore/Extensions/PrefixKeyVaultSecretManager.cs

```diff
@@ -0,0 +1,86 @@
+        /// <exception cref="ArgumentNullException">
+        /// The value of <paramref name="prefix"/> is <see langword="null"/>.
+        /// </exception>
+        /// <exception cref="ArgumentException">
+        /// The value of <paramref name="prefix"/> is empty string, or string containing only white-space characters.
+        /// </exception>
+        public PrefixKeyVaultSecretManager(string prefix)
+        {
+            Check.NotNullOrWhiteSpace(prefix, nameof(prefix));
+            AddPrefix(prefix);
+        }
+
+        /// <summary>
+        /// Creates an instance with specified <paramref name="prefixes"/>.
+        /// </summary>
+        /// <param name="prefixes"></param>
+        /// <exception cref="ArgumentNullException">
+        /// The value of <paramref name="prefixes"/> is <see langword="null"/>.
+        /// </exception>
+        /// <exception cref="ArgumentException"><paramref name="prefixes"/> is an ampty list, or does not contain any valid
+        /// prefix. Prefix must not be an empty string or a string containing only white-space characters.
+        /// </exception>
+        public PrefixKeyVaultSecretManager(IEnumerable<string> prefixes)
+        {
+            Check.NotNull(prefixes, nameof(prefixes));
+            foreach (string prefix in prefixes)
+            {
+                if (!string.IsNullOrWhiteSpace(prefix))
+                {
+                    AddPrefix(prefix);
+                }
+            }
+            if (_prefixes.Count == 0)
+            {
+                throw new ArgumentException("At least one valid prefix must be supplied. Prefix cannot be an empty string " +
+                    "or consists only of white-space characters.", nameof(prefixes));
+            }
+        }
+
+        private void AddPrefix(string prefix) => _prefixes.Add(prefix + "-");
+
+        /// <summary>
+        /// Checks if <see cref="KeyVaultSecret"/> value should be retrieved. The secret is retrieved, if its name
+        /// has one of the defined prefixes.
+        /// </summary>
+        /// <param name="secret">The <see cref="SecretProperties"/> instance.</param>
+        /// <returns><code>true</code> if secrets value should be loaded, otherwise <code>false</code>.</returns>
+        public override bool Load(SecretProperties secret)
+            => _prefixes.Any(prefix => secret.Name.StartsWith(prefix, StringComparison.OrdinalIgnoreCase));
+
+        /// <summary>
+        /// Maps secret to a configuration key. Prefix is removed from secret name and double dashes (<c>--</c>)
+        /// are replaced with colon (<c>:</c>).
+        /// </summary>
+        /// <param name="secret">The <see cref="KeyVaultSecret"/> instance.</param>
+        /// <returns>Configuration key name to store secret value.</returns>
+        public override string GetKey(KeyVaultSecret secret)
+        {
+            int keyStart = secret.Name.IndexOf('-') + 1;
```

Comment:
Podľa tohto ziavne prefix nemôže byť viacslovný oddelený a `-`. e.g.: `toto-je-moj-prefix`.
Som s tým v pohode, ale bolo by dobré to napísať do dokumentačného komentára tej metódy `AddAzureKeyVault`

Reply:
A asi aj test na to napísať.